crates/pathfinder/src/monitoring/metrics.rs

```diff
@@ -0,0 +1,80 @@
+pub mod middleware {
+    use jsonrpsee::core::middleware::Middleware;
+    use metrics::increment_counter;
+    use std::time::Instant;
+
+    #[derive(Debug, Clone)]
+    pub struct RpcMetricsMiddleware;
+
+    impl Middleware for RpcMetricsMiddleware {
+        type Instant = Instant;
+
+        fn on_request(&self) -> Self::Instant {
+            Instant::now()
+        }
+
+        fn on_call(&self, name: &str) {
+            increment_counter!(format!("{name}_calls_total"));
+        }
+    }
+
+    #[derive(Debug, Clone)]
+    pub enum MaybeRpcMetricsMiddleware {
+        Middleware(RpcMetricsMiddleware),
+        Noop,
+    }
+
+    impl jsonrpsee::core::middleware::Middleware for MaybeRpcMetricsMiddleware {
+        type Instant = std::time::Instant;
+
+        fn on_request(&self) -> Self::Instant {
+            std::time::Instant::now()
+        }
+
+        fn on_call(&self, name: &str) {
+            match self {
+                MaybeRpcMetricsMiddleware::Middleware(x) => x.on_call(name),
+                MaybeRpcMetricsMiddleware::Noop => {}
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod test {
+    use metrics::{Recorder, SetRecorderError};
+    use std::sync::{Arc, Mutex, MutexGuard};
+
+    lazy_static::lazy_static! {
+        static ref RECORDER_LOCK: Arc<Mutex<()>> = Arc::new(Mutex::new(()));
+    }
```

Comment:
```suggestion
    static RECORDER_LOCK: Mutex<()> = Mutex::new(());
```

We can have now static mutexes with 1.63, and since it's static, we don't have to have an Arc around it.

Though unsure if we have 1.62.1 still documented somewhere as MSRV. We should just replace those with "latest stable rust".

Reply:
TIL: static mutexes @ >=1.63